.github/workflows/clangformat.yml

```diff
@@ -1,26 +1,71 @@
 name: Clang Format
 
 on:
+  push:
+    branches: [ master, develop, clang-format ]
   pull_request:
     branches: [ master, develop ]
 
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write  # Needed to push a commit
+
     steps:
     - uses: actions/checkout@v4
 
     - name: Run clang-format against C++ files touched by the PR
+      if: ${{ github.event_name == 'pull_request' }}
       shell: bash
       run: |
           echo "GITHUB_REF=$GITHUB_REF GITHUB_BASE_REF=$GITHUB_BASE_REF GITHUB_HEAD_REF=$GITHUB_HEAD_REF"
           git fetch --all
           clang-format --version
           ./ci/clang-format.sh remotes/origin/$GITHUB_HEAD_REF remotes/origin/$GITHUB_BASE_REF
 
+    - name: Run clang-format for entire codebase
+      if: ${{ github.event_name == 'push' }}
+      shell: bash
+      run: |
+          clang-format --version
+          find . \( -name "*.hpp" -o -name "*.h" -o -name "*.hh" -o -name "*.cc" -o -name "*.cpp" -o -name "*.c" \) | xargs -P $(nproc) -I '{}' clang-format -style=file -i -fallback-style=none "{}"
+          # clang-format will auto correct files so prepare the diff and use this as artifact
+          git diff > clang_format.patch
+
+          # Delete if nothing otherwise exit 1 to indicate a failed job
+          if [ ! -s clang_format.patch ]; then
+            rm clang_format.patch
+            exit 0
+          else
+            echo "clang-format auto corrected files:"
+            git diff --name-only
+            echo -e "\nPlease correct these files. You can run ci/clang-format.sh locally and commit changes"
+            exit 1
+          fi
```

Comment:
When push to develop/master, it runs for the entire codenbase, which takes 1 min.